src/Domain/Encryption/Crypter.spec.ts

```diff
@@ -0,0 +1,62 @@
+import 'reflect-metadata'
+
+import { Crypter } from './Crypter'
+
+describe('Crypter', () => {
+  const encryptionIterations = 1000
+  const userServerKey = 'my-private-key'
+
+  const createCrypter = () => new Crypter(encryptionIterations)
+
+  it('should encrypt and decrypt data', async () => {
+    const sampleData = 'My-Super-Secret-Data'
+
+    const encryptedSampleData = await createCrypter().encrypt(Crypter.ENCRYPTION_VERSION_1, sampleData, userServerKey)
+
+    expect(encryptedSampleData).toHaveLength(156)
+
+    expect(
+      await createCrypter().decrypt(Crypter.ENCRYPTION_VERSION_1, encryptedSampleData, userServerKey)
+    ).toEqual('My-Super-Secret-Data')
+  })
+
+  it('should encrypt same data with different output every time', async () => {
+    const sampleData = 'My-Super-Secret-Data'
+
+    const encryptedSampleData1 = await createCrypter().encrypt(Crypter.ENCRYPTION_VERSION_1, sampleData, userServerKey)
+
+    expect(encryptedSampleData1).toHaveLength(156)
```

Comment:
Ideally this test would just have a single assertion: `expect(encryptedSampleData1).not.toEqual(encryptedSampleData2)`. The length thing is already validated with the previous test.